blockly-core/core/ui/contract_editor/x_button.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+goog.provide('Blockly.XButton');
+
+/**
+ * A DOM-based input which sends events on changes
+ * @param {!Object} options
+ *          onNameChanged (newName: String)
+ *          onEnterPressed
+ * @constructor
+ */
+Blockly.XButton = function (options) {
+  this.onButtonPressed = options.onButtonPressed;
+
+  // For disposal
+  this.eventsToUnbind_ = [];
+  this.buttonElement_ = null;
+};
+
+Blockly.XButton.prototype.render = function (parent) {
+  var buttonElement = goog.dom.createDom('button');
+  buttonElement.className = "btn";
+  buttonElement.innerHTML = "x";
+  buttonElement.style.marginRight = "-10px";
+
+  parent.appendChild(buttonElement);
+
+  this.eventsToUnbind_.push(Blockly.bindEvent_(buttonElement, 'mousedown', this, goog.bind(function () {
+    if (this.onButtonPressed) {
+      this.onButtonPressed();
+    }
```

Comment:
I've been seeing a lot of this sort of pattern (if function exists, call function) between this PR and the one I recently did for Brad.. What do we think of instead writing this as:
```
this.onButtonPressed && this.onButtonPressed();
```
Def more concise, I'm not sure if it makes it more or less clear.